### PR TITLE
Add RBAC tests for remotes

### PR DESCRIPTION
**Pull request by mdellweg**
This is not yet really testing much rbac related.
But it shows how to overcome the current inability to create users and assign permissions.

### CODE REVIEW

**Comment by lubosmj** [comment]
I am not reviewing this PR, but we will definitely benefit from it when it will be merged. Furthermore, we should also try to assign and remove particular object-level permissions for the user by utilizing https://django-guardian.readthedocs.io/en/stable/userguide/assign.html?highlight=assign_perm#assign-obj-perms to see how the code behaves.

**Comment by ipanova** [comment]
This looks good for a start, adding a second user with different set of perms would be a good course of actions to continue on.

**Comment by mdellweg** [comment]
I guess, adding object level permissions is a good thing we need to test, but we probably need a way to add those independently of the user creation process anyway. I'd leave that as an exercise for another PR.

File: pulp_container/tests/functional/api/test_rbac_remotes.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+class RBACRemotesTestCase(unittest.TestCase):
+    """RBAC remotes."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables and prepare api users."""
+        cls.user1 = gen_user(["container.add_containerremote"])
+        cls.user2 = gen_user(["container.view_containerremote"])
```
Comment (ipanova) [inline pulp_container/tests/functional/api/test_rbac_remotes.py:27]: does it makes sense to add one more test in this PR that checks user2 cannot create remotes?

File: pulp_container/tests/functional/api/test_rbac_remotes.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        # read without read permission
+        with self.assertRaises(ApiException):
+            remote = self.user3["remote_api"].read(self.remote.pulp_href)
+
+    @skip_if(bool, "remote", False)
+    def test_02_read_remotes(self):
+        """Read a remote by its name."""
+        page = self.user1["remote_api"].list(name=self.remote.name)
+        self.assertEqual(len(page.results), 1)
+        for key, val in self.remote.to_dict().items():
```
Comment (ipanova) [inline pulp_container/tests/functional/api/test_rbac_remotes.py:78]: how about skipping this? I seems like the main purpose is to confirm that the page has 1 result

File: pulp_container/tests/functional/api/test_rbac_remotes.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        body = gen_container_remote()
+        body["name"] = self.remote.name
+        with self.assertRaises(ApiException):
+            self.user1["remote_api"].create(body)
+
+    @skip_if(bool, "remote", False)
+    def test_02_read_remote(self):
+        """Read a remote by its href."""
+        remote = self.user1["remote_api"].read(self.remote.pulp_href)
+        for key, val in self.remote.to_dict().items():
```
Comment (ipanova) [inline pulp_container/tests/functional/api/test_rbac_remotes.py:64]: how about skipping this? It seems like the main purpose is to confirm that user can read a specific remote he looked for.

File: pulp_container/tests/functional/api/test_rbac_remotes.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        body = _gen_verbose_remote()
+        type(self).remote = self.user1["remote_api"].create(body)
+        for key in ("username", "password"):
+            del body[key]
+        for key, val in body.items():
+            with self.subTest(key=key):
+                self.assertEqual(self.remote.to_dict()[key], val, key)
+
+    @skip_if(bool, "remote", False)
+    def test_02_create_same_name(self):
```
Comment (ipanova) [inline pulp_container/tests/functional/api/test_rbac_remotes.py:49]: I am not entirely sure I see benefit in this exact test what would focus on the rbac testing itself, can you elaborate?

Comment (mdellweg) [reply pulp_container/tests/functional/api/test_rbac_remotes.py:49]: I must confess, i just took the tests from crud_remotes and added some rbac.
+1 to be more specific about what we test.

**Review by ipanova — APPROVED** [review]

**Pull request merged**